### DISCUSSION TITLE
Why is the btree_container's size_type signed?

### DISCUSSION THREAD

**Discussion by bobaoCrbs** [greg7mdp/parallel-hashmap#106] [Q&A] [answered]
Hi,
I am just wondering why the size() of btree containers is of type signed long int. (instead of just size_t?). 
i.e. [this line](https://github.com/greg7mdp/parallel-hashmap/blob/1f2671ff0a12710f19c92080693c857f918bda88/parallel_hashmap/btree.h#L840)
Thanks
Bo

**Comment by greg7mdp** [comment]
Hi @bobaoCrbs ,

Actually I don't know the answer. I'm sure there is a good reason, but I don't have time to look into it right now. Code was extracted from [abseil](https://github.com/abseil/abseil-cpp), with a few changes to make it header only. The Abseil developers could probably answer the question right away if you ask there.

Thanks!

**Reply by bobaoCrbs** [reply]
Thanks for replying, will ask in abseil repo

**Comment by bobaoCrbs** [comment, marked as answer]
Here is their answer: https://github.com/abseil/abseil-cpp/discussions/993

**Comment by greg7mdp** [comment]
Just checked in a change to make size_type unsigned. Thanks again for the input!
